microstream-cache/src/main/java/io/micronaut/microstream/cache/CacheConfigurationProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.microstream.cache;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.naming.Named;
+import one.microstream.cache.types.CacheConfiguration;
+
+/**
+ * @param <K> The key type
+ * @param <V> The value type
+ * @author Tim Yates
+ * @since 1.0.0
+ */
+public interface CacheConfigurationProvider<K, V> extends Named {
```

Comment:
What's the purpose of this interface? Are there going to be possibly other implementations?

Also, note that in micronaut-cache we usually suffix configuration classes with `*Configuration`. Some examples:

- https://github.com/micronaut-projects/micronaut-cache/blob/master/cache-ehcache/src/main/java/io/micronaut/cache/ehcache/configuration/EhcacheConfiguration.java
- https://github.com/micronaut-projects/micronaut-cache/blob/master/cache-infinispan/src/main/java/io/micronaut/cache/infinispan/InfinispanHotRodClientConfiguration.java
- https://github.com/micronaut-projects/micronaut-cache/blob/master/cache-hazelcast/src/main/java/io/micronaut/cache/hazelcast/HazelcastClientConfiguration.java

Reply:
We have removed this class. They Microstream cache configuration is a bit convoluted.